source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -157,6 +159,12 @@ bool PolicyMatcher::matches(const Network::Connection& connection,
          principals_.matches(connection, headers, metadata);
 }
 
+bool RequestedServerNameMatcher::matches(const Network::Connection& connection,
+                                         const Envoy::Http::HeaderMap&,
+                                         const envoy::api::v2::core::Metadata&) const {
+  return std::regex_match(connection.requestedServerName().data(), server_name_regex_);
```

Comment:
If the expectation is that this only applies to SNI, then this code could do the wrong thing, depending on your configured filter chain:

* If you don't explicitly add the tls_inspector filter, and you don't have any FilterChainMatch matching on servername, this field will be empty
* A different listener filter could set this to something other than the SNI value
* This could be a non-TLS connection, in which case there is no SNI value.

Given all this, I think you should document exactly what this is filtering on. Link to the docs on how requestedServerName() is set, give an example for how to set this up for TLS, and note that some filter chain configurations could lead to different results.

Reply:
Addressed in 92e54f4.